test_runner/regress/test_auth.py

```diff
@@ -82,6 +84,115 @@ def test_compute_auth_to_pageserver(neon_env_builder: NeonEnvBuilder):
             assert cur.fetchone() == (5000050000,)
 
 
+def test_pageserver_multiple_keys(neon_env_builder: NeonEnvBuilder):
+    neon_env_builder.auth_enabled = True
+    env = neon_env_builder.init_start()
+
+    pageserver_token_old = env.auth_keys.generate_pageserver_token()
+    pageserver_http_client_old = env.pageserver.http_client(pageserver_token_old)
+
+    pageserver_http_client_old.reload_auth_validation_keys()
+
+    env.auth_public_key_path = (
+        Path(env.repo_dir) / "auth_public_key.pem" / "auth_public_key_old.pem"
+    )
+    os.rename(
+        Path(env.repo_dir) / "auth_public_key.pem", Path(env.repo_dir) / "auth_public_key.pem.file"
+    )
+    os.mkdir(Path(env.repo_dir) / "auth_public_key.pem")
+    os.rename(
+        Path(env.repo_dir) / "auth_public_key.pem.file",
+        Path(env.repo_dir) / "auth_public_key.pem" / "auth_public_key_old.pem",
+    )
+
+    # Add a new key pair
+    env.regenerate_keys_at(
+        Path("auth_private_key.pem"), Path("auth_public_key.pem/auth_public_key_new.pem")
+    )
+
+    # Reload the keys on the pageserver side
+    pageserver_http_client_old.reload_auth_validation_keys()
+
+    # We can continue using the old token
+    pageserver_http_client_old.reload_auth_validation_keys()
```

Comment:
Is this duplicated on purpose?

Reply:
yes, it's there to ensure that authenticated http requests to the pageserver work (of which the reload function is one). maybe I should factor this out into a function to make it more clear.

Reply:
I've moved it into a new function in a subsequent commit.